### ISSUE TITLE
Inserting redundant loop changes image to vertical stripes

### ISSUE THREAD

**Issue by paulthomson** [mc-imperial/shader-compiler-bugs#31]
[[Shaders, screenshots, error log, reproduction steps, etc.]](https://github.com/mc-imperial/shader-compiler-bugs/tree/master/samsung-galaxy-s6/wrong_images/large-v100-b997a4e1b624027b_inv_variant_47) [[ZIP]](https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/mc-imperial/shader-compiler-bugs/tree/master/samsung-galaxy-s6/wrong_images/large-v100-b997a4e1b624027b_inv_variant_47)

This shader was very difficult to reduce. Even small changes cause the issue to disappear. As explained in the link above:

* A loop is inserted which calls a new function GLF_live13map. However, the loop does not access existing variables and the additional variables are dead after the loop, so it should have no effect. 

Perhaps local variables or registers get clobbered somehow. 

Recipient:

![recipient image](https://github.com/mc-imperial/shader-compiler-bugs/raw/master/samsung-galaxy-s6/wrong_images/large-v100-b997a4e1b624027b_inv_variant_47/recipient_reduced_final_manual.png "recipient image")

Variant:

![variant image](https://github.com/mc-imperial/shader-compiler-bugs/raw/master/samsung-galaxy-s6/wrong_images/large-v100-b997a4e1b624027b_inv_variant_47/variant_47_reduced_final_manual.png "variant image")

**Comment by paulthomson** [comment]
Reported:
https://community.arm.com/thread/10996

**Comment by paulthomson** [comment]
Fixed independently on newer drivers, although the Samsung Galaxy S6 currently does not use the newer drivers. 

The old forum has been deleted, but this was the response.

ARM:
We've confirmed that we can reproduce this one in all driver versions up to and including our r11p0 driver release from October last year. We've been unable to reproduce in any newer drivers (r12p0 onwards), so there is a bug-fixed production driver available to OEM device manufacturers to take if they are willing to push a driver update.

Also from a private email:
it looks like there was indeed an obscure register allocation bug in the driver from 2015, which has been fixed in later versions of our DDK.
